articles/article-088-no-free-lunch.md

```diff
@@ -0,0 +1,25 @@
+<!-- BEGIN TITLE -->
+# There is no such thing as a free lunch
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In Computer Science we have  a wide range of very different problems and accordingly many approaches to tackle them.
+We might for example try to find some 'optimal' solution for a search problem, but it is unclear which specific algorithm one should use.
+Hence, it is only natural to think about algorithms that could always provide us the best solution, regardless of the problems subtleties.
+
+However, even though a method might result in higher performance for one problem, this comes with the price of diminishing performance for other problems.
+This is formalized in the so-called 'No Free Lunch' theorems, which state that no single method works better than any other for all possible problems.
+In practice, it is therefore necessary to carefully select a method for the individual problem at hand.
+
+In other words, there is always a cost associated with selecting a method and unfortunately there is no such thing as a free lunch.
```

Comment:
unless you get a job at google apparently (yes I know it is technically not free, it was a bad attempt at a joke)